app/src/main/kotlin/com/waz/zclient/settings/account/color/AccentColorDialogFragment.kt

```
@@ -0,0 +1,46 @@
+package com.waz.zclient.settings.account.color
+
+import android.app.Dialog
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import com.waz.zclient.R
+import com.waz.zclient.core.extension.withArgs
+import kotlinx.android.synthetic.main.dialog_fragment_accent_color.view.*
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.InternalCoroutinesApi
+
+@InternalCoroutinesApi
+@ExperimentalCoroutinesApi
+class AccentColorDialogFragment : DialogFragment() {
+
+    private var listener: OnAccentColorChangedListener? = null
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+
+        val view = requireActivity().layoutInflater.inflate(R.layout.dialog_fragment_accent_color, null)
+
+        val initialColorId = arguments?.getInt(DEFAULT_ACCENT_COLOR_BUNDLE_KEY, DEFAULT_ACCENT_COLOR_ID)
+
+        val colors = resources.getIntArray(R.array.settings_accent_colors_ids).zip(resources.getIntArray(R.array.settings_accent_colors))
+            .map { AccentColor(it.first, it.second) }
+
+        initialColorId?.let {
```

I think we shouldn't give in nullability here. we're providing a newInstance method with arguments, and we're giving a default value, so if initialColorId is still null at this point, it's better to throw NPE rather than having some unexpected ui behaviour.

my suggestion is to change line 23 to:
**arguments!!**.getInt(DEFAULT_ACCENT_COLOR_BUNDLE_KEY, DEFAULT_ACCENT_COLOR_ID)

and getting rid of the `?.let` here